plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerSplitManager.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.faker;
+
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplitManager;
+import io.trino.spi.connector.ConnectorSplitSource;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.Constraint;
+import io.trino.spi.connector.DynamicFilter;
+import io.trino.spi.connector.FixedSplitSource;
+
+import java.util.List;
+
+public class FakerSplitManager
+        implements ConnectorSplitManager
+{
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            DynamicFilter dynamicFilter,
+            Constraint constraint)
+    {
+        return new FixedSplitSource(List.of(new FakerSplit()));
```

Comment:
This makes the generation for a table sequential. That's probably slow for large tables.
Maybe look at the limit threshold in the table handle and generate one split per 100K values.

Reply:
I removed the code that did this after https://github.com/trinodb/trino/pull/23691#discussion_r1789468846. Can we add this as a follow-up, or is it required to merge this?

Reply:
It would be good to handle it here so that generation of large data sets doesn't become sequential

Reply:
I added back the `min-splits` config option, but in a separate commit, until we agree on how to do this. You can also push commits to this branch.